src/glfw/window.cc

```diff
@@ -0,0 +1,233 @@
+{
+
+void
+setWindowHints()
+{
+    glfwDefaultWindowHints();
+
+    // clang-format off
+    glfwWindowHint( GLFW_RESIZABLE,               GLFW_TRUE );
+    glfwWindowHint( GLFW_VISIBLE,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FOCUSED,                 GLFW_TRUE );
+    glfwWindowHint( GLFW_FLOATING,                GLFW_FALSE );
+    glfwWindowHint( GLFW_TRANSPARENT_FRAMEBUFFER, GLFW_FALSE );
+    glfwWindowHint( GLFW_FOCUS_ON_SHOW,           GLFW_TRUE );
+    glfwWindowHint( GLFW_DOUBLEBUFFER,            GLFW_TRUE );
+    glfwWindowHint( GLFW_REFRESH_RATE,            GLFW_DONT_CARE );
+    glfwWindowHint( GLFW_CLIENT_API,              GLFW_NO_API );
+    // clang-format on
+} // setWindowHints
+
+std::string
+getHandleAddressString( const Window* window )
+{
+    return std::to_string( reinterpret_cast<uint64_t>( window ) );
+} // getHandleAddressString
+
+void
+logGLFWaction(
+    std::string&& action,
+    std::string&& handleAddressString = {},
+    std::string&& additional = {} //
+)
+{
+    std::string output = "GLFW " + action + ".";
+
+    if ( !handleAddressString.empty() )
+    {
+        // clang-format off
+        output += "\n"
+                  "Handle = " + handleAddressString;
+        // clang-format on
+
+        if ( !additional.empty() )
+        {
+            output += ", " + additional + ".";
+        }
+    }
+
+    spdlog::info( output );
+} // logGLFWaction
+
+} // namespace
+
+WindowType*
+Window::createWindow(
+    const WindowConfig& config,
+    Window* bound_handle //
+)
+{
+    setWindowHints();
+
+    WindowType* glfwWindow = glfwCreateWindow(
+        config.width,
+        config.height,
+        config.title,
+        config.fullscreen ? glfwGetPrimaryMonitor() : nullptr,
+        nullptr );
+
+    glfwSetWindowUserPointer( glfwWindow, bound_handle );
+    glfwSetFramebufferSizeCallback( glfwWindow, framebufferSizeCallback );
+
+    logGLFWaction(
+        "create window",
+        getHandleAddressString( bound_handle ) //
+    );
+
+    return glfwWindow;
+} // createWindow
+
+void
+Window::destroyWindow( WindowType* glfwWindow )
+{
+    Window* window = getHandle( glfwWindow );
+
+    glfwDestroyWindow( glfwWindow );
+
+    logGLFWaction(
+        "destroy window",
+        getHandleAddressString( window ) //
+    );
+} // destroyWindow
+
+void
+Window::framebufferSizeCallback(
+    WindowType* glfwWindow,
+    int width,
+    int height //
+)
+{
+    Window* window = getHandle( glfwWindow );
+
+    logGLFWaction(
+        "framebuffer resize",
+        getHandleAddressString( window ),
+        "width = " + std::to_string( width ) + ", height = " + std::to_string( height ) //
+    );
+
+    window->m_resize_callback( window, width, height );
+} // framebufferSizeCallback
+
+FramebufferSizeType
+Window::getFramebufferSize() const&
+{
+    int width = 0;
+    int height = 0;
+
+    glfwGetFramebufferSize( m_handle.get(), &width, &height );
+
+    return std::make_pair( width, height );
```

Comment:
This function has a return type ```FramebufferSizeType```. You should not use ```std::make_pair``` here. Prefer aggregate initialization:

```cpp
return { width, height };
```